hail/src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -979,15 +1040,15 @@ object PruneDeadFields {
         val bodyEnv = memoizeValueIR(body,
           requestedType.asInstanceOf[TStream].elementType,
           memo)
-        val valueTypes = names.zip(as).map { case (name, a) =>
+        val valueTypes = (names, as).zipped.map { case (name, a) =>
```

Comment:
As discussed in person, function literals with `case` still force a tuple to be allocated.